dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -109,4 +84,116 @@ def report_abuse_form
       age: current_user&.age,
     }
   end
+
+  # API routes, moved from legacy channels_api.rb.
+
+  # GET /v3/channels/:channel_id/abuse
+  # Get an abuse score.
+  def show_abuse
+    return head :unauthorized unless project_validator?
```

Comment:
our best practice is to perform these authorization checks using [CanCan](https://github.com/CanCanCommunity/cancancan). Here is one example: https://github.com/code-dot-org/code-dot-org/blob/7d8cb181e56f0ee28e563a4d42673df9f64ee91d/dashboard/app/models/ability.rb#L159-L160

Reply:
I added to ability.rb:

```ruby
can :destroy_abuse, :all if user.project_validator?
```

And in the controller action:

```ruby
return head :unauthorized unless can?(:destroy_abuse, :any)
```

And many variations of both, but it still returns false for the `can?` check no matter what. Am I missing anything obvious?

Reply:
Nevermind, apparently you can't call the same action multiple times in different user contexts in the same test, or cancan gets confused.

Reply:
ah, I had a vague memory that calling controller actions multiple times in one test case might not work. TIL it's because of CanCan! 💡

Reply:
you can also skip the various `return head :unauthorized unless can?` calls by calling `authorize_resource` once at the top of the controller file. this ensures that cancan is being checked for every controller method (unless otherwise specified). however, I think you've done enough here to get this logic into cancan and I think it is ok if you want to defer this for now.

Reply:
oh, I wasn't sure if I was supposed to use that pattern or not. good to know!